src/main/kotlin/three/consulting/epoc/dto/ProjectDTO.kt

```diff
@@ -3,26 +3,27 @@ package three.consulting.epoc.dto
 import three.consulting.epoc.entity.Project
 import java.time.LocalDate
 import java.time.LocalDateTime
+import javax.validation.constraints.NotBlank
 
-class ProjectDTO(
+data class ProjectDTO(
     val id: Long? = null,
-    val name: String,
+    @field:NotBlank val name: String,
     val description: String? = null,
-    val startingDate: LocalDate,
+    @field:NotBlank val startDate: LocalDate,
     val endDate: LocalDate? = null,
-    val customer: CustomerDTO? = null,
-    val managingEmployee: EmployeeDTO? = null,
+    @field:NotBlank val customer: CustomerDTO,
+    @field:NotBlank val managingEmployee: EmployeeDTO,
```

Comment:
`NotBlank` is for strings only. There are other validator too though, see https://www.baeldung.com/java-bean-validation-not-null-empty-blank.
But here only the name field is applicable for such.